signer/src/storage/model.rs

```diff
@@ -1352,6 +1349,260 @@ pub struct WithdrawalRejectEvent {
     pub signer_bitmap: BitArray<[u8; 16]>,
 }
 
+impl From<u8> for BitcoinBlockHeight {
+    fn from(value: u8) -> Self {
+        Self(value as u64)
+    }
+}
+impl From<u16> for BitcoinBlockHeight {
+    fn from(value: u16) -> Self {
+        Self(value as u64)
+    }
+}
+impl From<u32> for BitcoinBlockHeight {
+    fn from(value: u32) -> Self {
+        Self(value as u64)
+    }
+}
+impl From<u64> for BitcoinBlockHeight {
+    fn from(value: u64) -> Self {
+        Self(value)
+    }
+}
+impl From<usize> for BitcoinBlockHeight {
+    fn from(value: usize) -> Self {
+        Self(value as u64)
+    }
+}
+
+// Conversion u64 => StacksBlockHeight is not implemented intentionally.
+// Use deref instead.
+
```

Comment:
qq: why? Also s/StacksBlockHeight/BitcoinBlockHeight/

Reply:
In some cases where rust is not smart enough to understand how to interpret `.into()` it was much nicer looking to use deref:
`*height` vs `u64::from(height)`.

And I disabled this to make everything more consistent

Reply:
This conversion was actually requested by @cylewitruk here: https://github.com/stacks-sbtc/sbtc/pull/1448#discussion_r2026921374

For me personally I don't think it making me having weird feelngs, but I also agree that usage of u128 is not so wide, so I'm fine with both. Thus leaving this discussion to you two =)

Reply:
The deref to me is natural as it's deref'ing to its inner type, whereas for `u128` it's an actual conversion. I have nothing against also including `From<> for u64`'s, though. My argument for the `u128`conversion, despite its narrow use, are down [here](https://github.com/stacks-sbtc/sbtc/pull/1448#discussion_r2037175489).

Reply:
I'm happy with the comment specifying that there's nothing fishy going on, resolving this convo.